Sources/Workspace/Workspace.swift

```diff
@@ -4071,6 +4071,11 @@ extension Workspace {
             return nil
         }
 
+        // File-system based dependencies do not provide a custom file system object.
+        if case .fileSystem(_) = state {
+            return nil
+        }
+
         guard case .custom(_, _) = state else {
             observabilityScope.emit(error: "invalid managed dependency state for custom dependency: \(state)")
```

Comment:
should this throw instead of emitting a diagnostics? end result is the same I guess, but could terminate earlier with the error approach

Reply:
I think I prefer to keep these as diagnostics?